src/main/java/com/softserve/kickscooter/kickscootervehicle/management/controller/ManagementController.java

```diff
@@ -0,0 +1,55 @@
+package com.softserve.kickscooter.kickscootervehicle.management.controller;
+
+import com.softserve.kickscooter.kickscootervehicle.management.dto.ScooterCreateDto;
+import com.softserve.kickscooter.kickscootervehicle.management.dto.ScooterTechInfoDto;
+import com.softserve.kickscooter.kickscootervehicle.management.model.Scooter;
+import com.softserve.kickscooter.kickscootervehicle.management.service.ManagementService;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Optional;
+import java.util.UUID;
+
+@Slf4j
+@RestController
+@RequestMapping("/scooters")
+@AllArgsConstructor
+public class ManagementController {
+
+    private ManagementService service;
+
+    @GetMapping
+    public ResponseEntity<Iterable<ScooterTechInfoDto>> getAllScooterTechInfo(){
+        return ResponseEntity.ok(service.getAllScooterTechInfo());
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<ScooterTechInfoDto> getScooterTechInfo(@PathVariable UUID id){
+        log.info("Retrieving scooter by id = " + id );
+        Optional<ScooterTechInfoDto> infoDto = service.getScooterTechInfo(id);
+        return infoDto.map(ResponseEntity::ok).orElseGet(() -> ResponseEntity.notFound().build());
+    }
+
+    @PostMapping
+    public ResponseEntity<UUID> registerScooter(@RequestBody ScooterCreateDto dto){
+        Scooter scooter = service.registerScooter(dto);
+        UUID id = scooter.getId();
+        log.info("New scooter registred, uuid of new scooter {}", id);
+        return ResponseEntity.status(201).body(id);
+    }
+
+    @DeleteMapping("/id/{id}")
+    public ResponseEntity<Void> utilizeScooter(@PathVariable UUID id){
+        log.info("Utilizing scooter with uuid = {}", id );
+        Boolean result = service.utilizeScooter(id);
```

Comment:
wrapper types might be null. In such a case, NullPointerException will be thrown from the `if` during unboxing.

Reply:
Fixed